### ISSUE TITLE
nappgui may falsely report memory leaks at exit when linking with static libraries on windows

### ISSUE THREAD

**Issue by colugomusic** [frang75/nappgui_src#152]
In `bmem_win.c`, `_bmem_atexit()` is currently defined as:
```c
void _bmem_atexit(void)
{
#if defined(__MEMORY_AUDITOR__)
#if _WITH_CRTDBG
    _CrtDumpMemoryLeaks();
#endif
#else
    cassert(i_HEAP == NULL);
/*HeapDestroy(i_HEAP);*/
#endif
}
```
I have found that this will falsely report memory leaks at exit if your application is linking with a library which still has static objects hanging around at the point where nappgui emits the call to `_CrtDumpMemoryLeaks`. In my case it is the `Boost.process` library. I think the order in which static objects across different linked libraries are deallocated is not well defined so nappgui should probably remove this call.

**Comment by frang75** [comment]
Hi @colugomusic, thanks for using NAppGUI.

Indeed, with external libraries with static objects it is possible that `_CrtDumpMemoryLeaks()` detects memory leaks due to objects not yet released. However, this call is very useful during development, since native Win32 objects are not controlled by the NAppGUI memory auditor. At the moment, it is not going to be removed.

Control over the CRT (`#include <crtdbg.h>`) is only included in `DEBUG` builds. You can also disable it by commenting the line:

`#define _WITH_CRTDBG 1`

**Comment by colugomusic** [comment]
> Hi @colugomusic, thanks for using NAppGUI.
> 
> Indeed, with external libraries with static objects it is possible that `_CrtDumpMemoryLeaks()` detects memory leaks due to objects not yet released. However, this call is very useful during development, since native Win32 objects are not controlled by the NAppGUI memory auditor. At the moment, it is not going to be removed.
> 
> Control over the CRT (`#include <crtdbg.h>`) is only included in `DEBUG` builds. You can also disable it by commenting the line:
> 
> `#define _WITH_CRTDBG 1`

Maybe a compromise could be to allow the user to opt out by passing an additional compiler flag, e.g. `_NAPPGUI_NO_CRTDBG` if they know they are linking against other libraries:
```c
#if defined(_MSC_VER) && _MSC_VER > 1400 && !defined(_NAPPGUI_NO_CRTDBG)
#define _WITH_CRTDBG 1
#include <crtdbg.h>
#endif
```
That way they don't have to keep a separate fork of nappgui.

**Comment by frang75** [comment]
@colugomusic 
Fixed in this commit: https://github.com/frang75/nappgui_src/commit/b65fc3194b22b77bab00ba079e25de96a4d3accc
Doc: https://nappgui.com/docs/r5452/en/guide/build.html#h3

**Comment by colugomusic** [comment]
> @colugomusic Fixed in this commit: [b65fc31](https://github.com/frang75/nappgui_src/commit/b65fc3194b22b77bab00ba079e25de96a4d3accc) Doc: https://nappgui.com/docs/r5452/en/guide/build.html#h3

Perfect thank you!

**Comment by frang75** [comment]
There is a typo in doc. The correct option is `-DCMAKE_DISABLE_CRTDBG=YES`
